---
title: 【Next.js】エラー画面を表示する方法
tags:
  - Next.js
  - Next.js13
private: false
updated_at: '2023-10-28T18:50:40+09:00'
id: aa72caf931576a59b239
organization_url_name: null
slide: false
ignorePublish: false
---
## ページを作成する

`page.tsx`を以下のように作成します。

```tsx:page.tsx
async function getData() {
  const res = await fetch('https://api.example.com/...')
 
  if (!res.ok) {
    throw new Error('エラーが発生しました。')
  }
 
  return res.json()
}
 
export default async function Page() {
  const data = await getData()
 
  return <main></main>
}
```

## エラー画面を作成する

`pages.tsx`と同じ階層に`error.tsx`を作成します。

```tsx:error.tsx
'use client'
 
import { useEffect } from 'react'
 
export default function Error({
  error,
  reset,
}: {
  error: Error & { digest?: string }
  reset: () => void
}) {
  useEffect(() => {
    console.error(error)
  }, [error])
 
  return (
    <div>
      <h2>{error.message}</h2>
      <button onClick={() => reset()}>
        Try again
      </button>
    </div>
  )
}
```

上記のように記述すると`page.tsx`の`throw new Error('エラーが発生しました。')`が実行された際に`エラーが発生しました。`というメッセージと再実行するためのボタンが表示されます。


:::note warn
`error.tsx`の`error.message`は本番環境では`An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.`というメッセージになります。
:::

:::note warn
`error.tsx`と同じ階層に`not-found.tsx`が存在しない場合に`page.tsx`で`notFound()`を実行すると`not-found.tsx`が表示されるのではなく、`error.tsx`の`error.message`に`NEXT_NOT_FOUND`が渡されて表示されます。
:::
